web/modules/custom/ilr_registrations/src/SerializedOrderManager.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Drupal\ilr_registrations;
+
+use Drupal\Core\Entity\EntityManagerInterface;
+use Drupal\commerce_order\Entity\OrderInterface;
+
+/**
+ * Class SerializedOrderService.
+ */
+class SerializedOrderManager implements SerializedOrderManagerInterface {
+
+  const POS_ID = 'register.ilr.cornell.edu';
+
+  /**
+   * Drupal\Core\Entity\EntityManagerInterface definition.
+   *
+   * @var \Drupal\Core\Entity\EntityManagerInterface
+   */
+  protected $entityManager;
+
+  /**
+   * Constructs a new SerializedOrderService object.
+   */
+  public function __construct(EntityManagerInterface $entity_manager) {
+    $this->entityManager = $entity_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getObjectForOrder(OrderInterface $order) {
+    $promotion_storage = $this->entityManager->getStorage('commerce_promotion');
+    $sf_mapping_storage = $this->entityManager->getStorage('salesforce_mapped_object');
+
+    $items = $order->getItems();
+    $customer = $order->getCustomer();
+    $billing_profile = $order->getBillingProfile();
+    $payments = $this->entityManager->getStorage('commerce_payment')->loadByProperties([
+      'order_id' => $order->id(),
+    ]);
+    $payment = reset($payments);
+    $payment_gateway = $payment->getPaymentGateway();
+
+    // Note that this function makes a remote call to the FreedomPay API.
+    // @todo Deal with possible remote transaction retrieval failure.
+    $transaction = $payment_gateway->getPlugin()->getTransaction($payment->getRemoteId());
+
+
+    $response = [
+      "point_of_sale" => $this::POS_ID,
+      "order_id" => $order->id(),
+      "payments" => [
+        [
+          "payment_type" => $payment_gateway->getPluginId(), //"freedompay_cc",
+          "payment_id" => $payment->id(),
+          "amount" => (float) $payment->getAmount()->getNumber(),
+          "transaction_id" => $payment->getRemoteId(),
+          "transaction_data" => $transaction,
+        ]
+      ],
+      "customer" => [
+        "contact_sfid" => null,
```

Comment:
Add a `@todo` here and in the participant loop regarding the contact sfid. That's work we could probably do without Dan's help using the native salesforce api to check for a contact with the email address of a given customer/participant email.